### ISSUE TITLE
errorType: 0 / reason: The magic number in GZip header is not correct. Make sure you are passing in a GZip stream. / httpStatusCode: NotFound / exception: System.IO.InvalidDataException: The magic number in GZip header is not correct. Make sure you are passing in a GZip stream.

### ISSUE THREAD

**Issue by santhosh250595** [kawansoft/AceQL.Client#4]
Hello team,

I'm getting the above error sometimes(not at the same point) when I execute ExecuteReaderAsync for a select query using postgres. If I retry the same again it'll work fine. Below is the detailed exception message I got. Please take a look and suggest.

**Make sure you are passing in a GZip stream. / httpStatusCode: NotFound / exception: System.IO.InvalidDataException: The magic number in GZip header is not correct. Make sure you are passing in a GZip stream.
   at System.IO.Compression.GZipDecoder.ReadHeader(InputBuffer input)
   at System.IO.Compression.Inflater.Decode()
   at System.IO.Compression.Inflater.Inflate(Byte[] bytes, Int32 offset, Int32 length)
   at System.IO.Compression.DeflateStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.IO.Compression.GZipStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.IO.Stream.InternalCopyTo(Stream destination, Int32 bufferSize)
   at System.IO.Stream.CopyTo(Stream destination)
   at AceQL.Client.Api.AceQLCommand.<ExecuteQueryAsStatementAsync>d__18.MoveNext()
2020-07-17 05:57:09,425 ERROR DBApiWrapper.DBManagement - stack    at AceQL.Client.Api.AceQLCommand.<ExecuteQueryAsStatementAsync>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at AceQL.Client.Api.AceQLCommand.<ExecuteReaderAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()**

**Comment by santhosh250595** [comment]
Hello,

Can we do kind of retry or some validation in the below part

**using (GZipStream decompressionStream = new GZipStream(input, CompressionMode.Decompress))
                        {
                            using (var stream = await file.OpenAsync(PCLStorage.FileAccess.ReadAndWrite).ConfigureAwait(false))
                            {
                                decompressionStream.CopyTo(stream);
                            }
                        }**

**Comment by ndepomereu** [comment]
Hi,
For now, Just set `AceQLConnection.GzipResult `property to false before first SQL query, and no compression will be done

```
            using (AceQLConnection connection = new AceQLConnection(connectionString))
            {
                connection.GzipResult = false;
                await ExecuteExample(connection).ConfigureAwait(false);
            }
```

I will laster check how to implement automatic retry if the stream is compressed, not sure if feasible and how to do it cleanly...

**Comment by ndepomereu** [comment]
This is marked close for now, as the proposed workaround will avoid producing a GZip stream..

**Comment by ndepomereu** [comment]
Hi,
We reopen this issue just to say there is a new `AceQLConnection.RequestRetry` property that allows forcing a retry if a 
SELECT query failed. If the failure was due to a GZIP issue, the failed request is retried without any compression. 

This is experimental and `AceQLConnection.RequestRetry` default value is `false`.

**Comment by GalTzemach** [comment]
The new `AceQLConnection.RequestRetry` property avilable from version 5.5.0?

**Comment by ndepomereu** [comment]
Yes

**Comment by GalTzemach** [comment]
Thanks

**Comment by ndepomereu** [comment]
I close as this issue is solved in v5.5.0.
